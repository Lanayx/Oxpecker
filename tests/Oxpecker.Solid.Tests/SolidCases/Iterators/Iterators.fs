module Oxpecker.Solid.Tests.Cases.Iterators

open Oxpecker.Solid

[<SolidComponent>]
let Test () =
    div() {
        For(each = [|"one"; "two"; "three"|]) {
            yield fun item index ->
                h2(id = (index() |> string)) {
                    item
                }
        }
        Index(each = [|"one"; "two"; "three"|]) {
            yield fun item index ->
                h2(id = (index |> string)) {
                    item()
                }
        }
    }
