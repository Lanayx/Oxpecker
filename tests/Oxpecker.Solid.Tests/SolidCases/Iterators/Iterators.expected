import { int32ToString } from "./fable_modules/fable-library-js.4.21.0/Util.js";

export function Test() {
    return <div>
        <For each={["one", "two", "three"]}>
            {(item, index) => <h2 id={int32ToString(index())}>
                {item}
            </h2>}
        </For>
        <Index each={["one", "two", "three"]}>
            {(item_1, index_1) => <h2 id={int32ToString(index_1)}>
                {item_1()}
            </h2>}
        </Index>
    </div>;
}

